File: Arrays/Two_Number_Sum/Java/twoNumSumSets.java
import java.util.*;

class twoNumSumSets {
  public static void main(String[] args) {
    int [] numbers = new int[] {1, 2, 4, 5, -5, -2};
    int targetSum = 2;

    twoNumSum(numbers, targetSum);
  };

  public static int[] twoNumSum(int[] numbers, int targetSum) {
    Set<Integer> nums = new HashSet<>();
    for (int num : numbers) {
      int potentialMatch = targetSum - num;
      if (nums.contains(potentialMatch)) {
        int[] result = new int[] {num, potentialMatch};
        System.out.println(Arrays.toString(result));
        System.out.println("You got the answer!");
        return result;
      } else {
        nums.add(num);
      }
    } 
    return new int[0];
  }
};

File: Arrays/Two_Number_Sum/TypeScript/twoNumSumSets.ts
export function twoNumSumSets(numbers: number[], targetSum: number) {
  const matchingSets = new Set();

  for (let num of numbers) {
    const potentialMatch: number = targetSum - num;

    if (matchingSets.has(potentialMatch)) {
      console.log([num, potentialMatch]);
      console.log("You got it!");
      return [num, potentialMatch];
    } else {
      matchingSets.add(num);
    }
  }
  return [];
}


let numbers: number[] = [1, 2, 4, 5, -5, -2];
let targetSum: number = 2;

twoNumSumSets(numbers, targetSum);
